From: Daniel Baumann <mail@daniel-baumann.ch>
Date: Fri, 29 Jun 2012 21:01:32 +0200
Subject: Use keyctl from the current build rather the one from system

Bug-Debian: http://bugs.debian.org/659299
Forwarded: no
Last-Update: 2014-05-18

Index: keyutils-1.5.9/tests/prepare.inc.sh
===================================================================
--- keyutils-1.5.9.orig/tests/prepare.inc.sh
+++ keyutils-1.5.9/tests/prepare.inc.sh
@@ -1,10 +1,12 @@
 # preparation script for running keyring tests
 
+alias keyctl="../../../../keyctl"
+
 # --- need to run in own session keyring
 if [ "x`keyctl rdescribe @s | sed 's/.*;//'`" != "xRHTS/keyctl/$$" ]
 then
     echo "Running with session keyring RHTS/keyctl/$$"
-    exec keyctl session "RHTS/keyctl/$$" sh $0 $@ || exit 8
+    keyctl session "RHTS/keyctl/$$" sh $0 $@ || exit 8
 fi
 
 # Set up for the Red Hat Test System
